Source/SwiftLintFramework/Rules/Lint/EmptyMethodOverrideRule.swift

```diff
@@ -0,0 +1,172 @@
+        """
+        override func viewDidDisappear(_ animated: Bool) {
+            childViewController.viewDidDisappear(animated)
+        }
+        """,
+        """
+        override func viewDidDisappear(_ animated: Bool) {
+            super.viewDidDisappear(animated)
+            print("View controller did disappear")
+        }
+        """,
+        """
+        public override init() {
+            super.init()
+        }
+        """,
+        """
+        override func setUp() {
+            super.setUp()
+            urlString = "https://httpbin.org/basic-auth"
+        }
+        """
+    ].map(wrapInClass)
+
+    static let triggeringExamples = [
+        """
+        override func a(){/*comment*/super.a()}
+        """,
+        """
+        override func viewDidLoad() {
+            super.viewDidLoad()
+
+            // Do any additional setup after loading the view.
+        }
+        """,
+        """
+        override func didReceiveMemoryWarning() {
+            super.didReceiveMemoryWarning()
+            // Dispose of any resources that can be recreated.
+        }
+        """,
+        """
+        override func becomeFirstResponder() -> Bool {
+            return super.becomeFirstResponder()
+        }
+        """,
+        """
+        internal
+        class
+        override
+        func setUp() {
+            super.setUp()
+        }
+        """
+    ].map(wrapInClass)
+
+    static let corrections = Dictionary(
+        uniqueKeysWithValues: EmptyMethodOverrideRule.triggeringExamples.map { ($0, wrapInClass("")) }
+    )
+
+    public static let description = RuleDescription(
+        identifier: "empty_method_override",
+        name: "Empty Method Override",
+        description: "Methods that don't do anything but call `super` can be removed",
+        kind: .lint,
+        nonTriggeringExamples: EmptyMethodOverrideRule.nonTriggeringExamples,
+        triggeringExamples: EmptyMethodOverrideRule.triggeringExamples,
+        corrections: EmptyMethodOverrideRule.corrections
+    )
+
+    public func violationRanges(
+        in file: File,
+        kind: SwiftDeclarationKind,
+        dictionary: [String: SourceKitRepresentable]
+    ) -> [NSRange] {
+        let overridingKinds: [SwiftDeclarationKind] = [
+            .functionMethodInstance,
+            .functionMethodClass,
+            .functionMethodStatic
+        ]
+
+        guard
+            overridingKinds.contains(kind),
+            dictionary.enclosedSwiftAttributes.contains(.override),
+            !dictionary.enclosedSwiftAttributes.contains(.public),
+            file.onlyCallsSuper(dictionary),
+            let offset = dictionary.offset,
+            let length = dictionary.length
+        else { return [] }
+
+        let startIndex = dictionary.swiftAttributes
+            .compactMap { $0.offset }
+            .min() ?? offset
+        let endIndex = offset + length
+
+        return [NSRange(startIndex..<endIndex)]
```

Comment:
nit: you can use `NSRange(location:length)` to avoid the need for `endIndex`

Reply:
Same as [here](https://github.com/realm/SwiftLint/pull/2680#discussion_r288588180), the `offset` variable isn't really the start of the violation, so I think it's more convenient to use the `endIndex` here for ensuring the correct violation range.

Reply:
but does this actually work? won't this be missing the length of the attributes? because `length` should be the length from `func` to the last `}`.

Reply:
Yes, the length of the attributes is why I'm using this NSRange initializer:

```swift
internal override func a() { super.a() }
^ 1               ^ 2                  ^ 3

1 startIndex
2 offset
3 endIndex (offset + length)
```

So the range from `startIndex` to `endIndex` is the one that should be removed. `NSRange(location:length)` would require the knowledge of how long the attributes (`internal` + `override`) are, and that seems more complicated to me.